packages/node/src/client.ts

```
@@ -1,57 +1,204 @@
+    this.INTEGRATION_ID = integrationId
+    this.BEARER_API_KEY = BEARER_API_KEY
+    this.clientOptions = options
+
+    this.client = axios
+  }
+
+  public auth = ({ authId, setupId }: { authId?: string; setupId?: string }) => {
+    this.AUTH_ID = authId || ''
+    this.SETUP_ID = setupId || ''
+
+    return this
+  }
+
+  public authenticate = this.auth // Alias
+
+  /**
+   * HTTP methods
+   */
+
+  public get = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('GET', endpoint, parameters, options)
+  }
+
+  public head = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('HEAD', endpoint, parameters, options)
+  }
+
+  public post = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('POST', endpoint, parameters, options)
+  }
+
+  public put = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('PUT', endpoint, parameters, options)
+  }
+
+  public options = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('OPTIONS', endpoint, parameters, options)
+  }
 
-  constructor(protected readonly token: string, clientOptions: Partial<TClientOptions> = {}) {
-    if (!token) {
-      throw new InvalidAPIKey(token)
+  public delete = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('DELETE', endpoint, parameters, options)
+  }
+
+  public patch = (endpoint: string, parameters?: BearerRequestParameters, options?: any) => {
+    return this.makeRequest('PATCH', endpoint, parameters, options)
+  }
+
+  protected makeRequest = (
+    method: BearerRequestMethod,
+    endpoint: string,
+    parameters?: BearerRequestParameters,
+    options?: BearerRequestOptions
+  ) => {
+    if (!this.BEARER_API_KEY) {
```

Let's move this into the constructor so that it fails fast rather than waiting the request to be performed